### PR TITLE
SSL logging and other improvements

### PR DESCRIPTION
**Pull request by Ashod**
- browser: protect against undefined keyEventFn
- wsd: always include the BIO errors in SSL logs
- wsd: trace poll events and revents flags arround ppoll
- wsd: test: correctly show the failing code in the assertion message
- wsd: minor improvements to StringVector::equals overloads
- wsd: exclude deprecated openssl API

### CODE REVIEW

File: net/Socket.hpp [hunk 1]
```diff
@@ -1117,15 +1117,18 @@ class StreamSocket : public Socket,
...
-                LOG_TRC('#' << getFD() << ": Read incoming data " << len << " bytes in addition to "
-                            << _inBuffer.size() << " buffered bytes ("
-                            << Util::symbolicErrno(last_errno) << ": " << std::strerror(last_errno)
-                            << ')');
-
-                if (len < 0 && last_errno != EAGAIN && last_errno != EWOULDBLOCK)
-                    LOG_SYS_ERRNO(last_errno, '#' << getFD() << ": Socket read returned " << len);
-            }
-            while (len < 0 && last_errno == EINTR);
+                if (len < 0)
```
Comment (mmeeks) [inline net/Socket.hpp:1122]: I really think EAGAIN and EWOULDBLOCK are too noisy to log but ... if you need them for a bit - why not.

File: net/SslSocket.hpp [hunk 2]
```diff
@@ -19,7 +20,7 @@
 class SslStreamSocket final : public StreamSocket
 {
 public:
-    SslStreamSocket(const std::string hostname, const int fd, bool isClient,
+    SslStreamSocket(std::string hostname, const int fd, bool isClient,
```
Comment (mmeeks) [inline net/SslSocket.hpp:23]: de-const-ipating hostname ?

**Review by mmeeks — APPROVED** [review]
Looks fine to me - minor comments =) thanks Ash.

File: net/Socket.hpp (hunk 1 above)
Comment (Ashod) [reply net/Socket.hpp:1122]: We did log them previously. This reduces the total log entries, actually.

File: net/SslSocket.hpp (hunk 2 above)
Comment (Ashod) [reply net/SslSocket.hpp:23]: Haha. Actually, the argument is moved, so the const doesn't help here. Clang-tidy fix.

**Pull request merged**